File: Python/Kyu8/Finish Guess the Number Game.py
'''
|_______________________________________________________________________________
| @Solver         : GziXnine.
| @Link           : https://www.codewars.com/kata/568018a64f35f0c613000054
| @description    : Implement a guessing game with a limited number of lives.
|                   - If the user guesses right, return True.
|                   - If the user guesses wrong, return False and deduct a life.
|                   - If the user exceeds the limit of lives, throw an error.
|_______________________________________________________________________________
'''
class Guesser:
  def __init__(self, number, lives):
    self.number = number
    self.lives = lives
  def guess(self, n):
    if self.lives < 1: raise "Too many guesses!"
    if self.number == n: return True
    self.lives -= 1
    return False

File: JavaScript/Kyu8/Finish Guess the Number Game.js
/*
|_______________________________________________________________________________
| @Solver         : GziXnine.
| @Link           : https://www.codewars.com/kata/568018a64f35f0c613000054
| @description    : A class for a guessing game with a limited number of lives.
|                   Throws an error if the user exceeds the guess limit.
|_______________________________________________________________________________
*/
class Guesser {
  constructor(number, lives) {
    this.number = number;
    this.lives = lives;
  }
  
  guess(n) {
    if (this.lives < 1) throw 'already dead'; // Throw an error if no lives left
    return (this.number === n || !(this.lives--)); // Return true for a correct guess, false otherwise
  }
}
